### ISSUE TITLE
kubefed join breaks with aws cluster

### ISSUE THREAD

**Issue by henriquetruta** [kubernetes/kubernetes#49041]
Joining a cluster from AWS fails in kubefed 1.7. The AWS cluster is also k8s 1.7
The federation is running in a GKE cluster 1.7.0. 

```
$ kubefed join aws --host-cluster-context=gke --context=fed
Error from server (Forbidden): clusterroles.rbac.authorization.k8s.io "federation-controller-manager:fed-aws-gke" is forbidden: attempt to grant extra privileges: [PolicyRule{Resources:["*"], APIGroups:["*"], Verbs:["*"]} PolicyRule{NonResourceURLs:["/healthz"], Verbs:["get"]}] user=&{admin admin [system:authenticated] map[]} ownerrules=[] ruleResolutionErrors=[]
```

This same command works in kubefed 1.6.6
I'm able to access the AWS cluster using its context.

/kind bug

**How to reproduce it (as minimally and precisely as possible)**:
1 - Create a GKE cluster
2 - Create an AWS cluster (I used conjure-up)
3 - kubefed (1.7) init on GKE cluster
4 - Join the AWS cluster
5 - It should fail

If kubefed 1.6 is used, it works.
Joining a GKE or On-prem cluster also works as expected

**Comment by quinton-hoole** [comment]
cc @madhusudancs

**Comment by GabrielSVinha** [comment]
I believe this is solved in 1.7+ where RBAC is default for all providers.

**Comment by irfanurrehman** [comment]
This issue was labelled only for sig/multicluster and is thus moved over to kubernetes/federation#23.
If this does not seem to be right, please reopen this and notify us @kubernetes/sig-multicluster-misc.
/close
